docs/development.rst

```diff
@@ -0,0 +1,497 @@
+                msg = 'Could not log into the world!'
+                raise exceptions.UnrecoverableActorFailure(msg)
+
+            # Make sure to support DRY mode all the time!
+            if self._dry:
+                self.log.info('Would have said Hi to %s' % world)
+                raise gen.Return()
+
+            # Finally, send the message!
+            try:
+                res = yield self._hello_world.send(
+                    from=name, to=world)
+            except WalkingAlone as e:
+                # Lets say that this error is completely un-handleable exception,
+                # there's no one to say hello to!
+                self.log.critical('Some extra information about this error...')
+
+                # Now, raise an exception that is will stop execution of Kingpin,
+                # regardless of the warn_on_failure setting.
+                raise exceptions.UnrecoverableActorException('Oh my: %s' % e)
+
+            # Return the value back to the execute method
+            raise gen.Return(res)
+
+        # The meat of the work happens in the _execute() method. This method
+        # is called by the BaseActor.execute() method. Your method must be
+        # wrapped in a gen.Coroutine wrapper. Note, the _execute() method takes
+        # no arguments, all arguments for the acter were passed in to the
+        # __init__() method.
+        @gen.coroutine
+        def _execute(self):
+            self.log.debug('Warming up the HelloWorld Actor')
+            
+            # Fire off an async request to a our private method for sending
+            # hello world messages. Get the response and evaluate
+            res = yield self._send_message(
+                self.option('name'), self.option('world')) 
+
+            # Got a response. Did our message really go through though?
+            if not res:
+                # The world refuses to hear our message... A shame, really, but
+                # not entirely critical.
+                self.log.error('We failed to get our message out ... just '
+                               'letting you know!')
+                raise exceptions.RecoverableActorFailure(
+                    'A shame, but I suppose they can listen to what they want')
+
+            # We've been heard!
+            self.log.info('%s people have heard our message!' % res)
+
+            # Indicate to Tornado that we're done with our execution.
+            raise gen.Return()
+
+Required Options
+^^^^^^^^^^^^^^^^
+
+The following options are baked into our *BaseActor* model and must be
+supported by any actor that subclasses it. They are fundamentally critical to
+the behavior of Kingpin, and should not be bypassed or ignored.
+
+``desc``
+''''''''
+
+A string describing the stage or action thats occuring. Meant to be human
+readable and useful for logging. You do not need to do anything intentinally to
+support this option (its handled in *kingpin.actors.base.BaseActor.\ **init**\ ()*).
```

Comment:
Formatting looks weird with the `*`s; should probably be a link to the docs. `it's` should contain an apostrophe